### PR TITLE
Update With More and Better Moving Method 

### PR DESCRIPTION
**Pull request by Omaranwa**

### CODE REVIEW

File: Assets/Scripts/Grid.cs [hunk 1]
```diff
@@ -41,52 +45,94 @@ public int maxSize
...
         Vector3 worldBottomLeft = transform.position - Vector3.right * gridWorldSize.x / 2 - Vector3.forward * gridWorldSize.y / 2;
 
-        
         for (int x = 0; x < gridSizeX; x++)
         {
             for (int y = 0; y < gridSizeY; y++)
             {
-                Vector3 worldPoint = worldBottomLeft + Vector3.right * (x * nodeDiamter + nodeRadius) + Vector3.forward * (y * nodeDiamter + nodeRadius);
-                bool walkble = !(Physics.CheckSphere(worldPoint, nodeRadius, unWalkableMask));
+                Vector3 worldPoint = worldBottomLeft + Vector3.right * (x * nodeDiameter + nodeRadius) + Vector3.forward * (y * nodeDiameter + nodeRadius);
```
Comment (zmeriksen) [inline Assets/Scripts/Grid.cs]: You can calculate this part in the first loop:
`worldBottomLeft + Vector3.right * (x * nodeDiameter + nodeRadius) + Vector3.forward`

Comment (zmeriksen) [reply Assets/Scripts/Grid.cs]: If I understand correctly... You should only need to calculate `(y * nodeDiameter + nodeRadius)` in the inner loop

Comment (Omaranwa) [reply Assets/Scripts/Grid.cs]: what about this line?
`bool walkble = !(Physics.CheckSphere(worldPoint, nodeRadius, unWalkableMask));`

Comment (Omaranwa) [reply Assets/Scripts/Grid.cs]: I will try to add it!

**Pull request merged**